storage/storage.go

```diff
@@ -284,23 +284,37 @@ type bucketBoundHostname struct {
 	hostname string
 }
 
-func (s pathStyle) host(bucket string) string {
+func (s pathStyle) host(endpoint, bucket string) string {
+	if endpoint != "" {
+		return stripScheme(endpoint)
+	}
+
+	// This check is needed for clientless calls to SignedURL
```

Comment:
This comment is a bit confusing, I guess you are saying that endpoint will be set to STORAGE_EMULATOR_HOST already if called via the bucketHandle method? If so probably not necessary to mention.

Reply:
Yes, that is what the comment implies - this check is only needed for using the standalone func, because the endpoint would be set already for the bucket method. But you're right - we have tests that will fail if the check is removed; will remove comment.